### PR TITLE
[Feature-request][Backend] Added Option for JSON logging

### LINKED ISSUE #6982: [Feature][Backend] Option for JSON logging
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Use case

As an operator, I want to be able to have devlake backend logs output in JSON format

### Description

Devlake backend is using `github.com/sirupsen/logrus` for logging and has hardcoded the `logrus.TextFormatter`.  Logrus also has an option to use a `JSONFormatter` https://pkg.go.dev/github.com/sirupsen/logrus#JSONFormatter.  Would be nice to have the option to get logs in JSON format.

Could be exposed via environment variable `LOGGING_FORMAT=json` with a default value of `text`.

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hanshal101**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [ ] I have added relevant tests.
- [ ] I have added relevant documentation.
- [ ] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

### Summary
What does this PR do?
This PR enables the Devlake backend logs output in JSON format

### Does this close any open issues?
Closes #6982 
Fixes #6982

### CODE REVIEW

File: backend/impls/logruslog/init.go [hunk 1]
```diff
@@ -47,10 +47,18 @@ func init() {
...
+		inner.SetFormatter(&logrus.JSONFormatter{
+			TimestampFormat: time.DateTime,
+		})
+	} else {
+		inner.SetFormatter(&logrus.TextFormatter{
+			TimestampFormat: time.DateTime,
+			FullTimestamp:   true,
+		})
+	}
+
```
Comment (ferminhg) [inline backend/impls/logruslog/init.go:61]: 🚀 Great work on streamlining the logging setup!

By adopting a switch statement, we not only align with the Go philosophy of simplicity and readability but also 🛠️ enhance our code's flexibility. 

This approach makes it super easy to introduce new formatters in the future.

📈 Adding or modifying loggers will be a breeze, ensuring our project remains adaptable and maintainable.

```suggestion
    var formatter logrus.Formatter
    format := os.Getenv("LOGGING_FORMAT")
	switch format {
	case "json":
		formatter = &logrus.JSONFormatter{
			TimestampFormat: time.DateTime,
		}
	default:
		formatter = &logrus.TextFormatter{
			TimestampFormat: time.DateTime,
			FullTimestamp:   true,
		}
	}
	inner.SetFormatter(formatter)
```

You could extract it to a method if you want to add readability:

```golang
func configureLogging(inner *logrus.Logger) {
	var formatter logrus.Formatter

	format := os.Getenv("LOGGING_FORMAT")
	switch format {
	case "json":
		formatter = &logrus.JSONFormatter{
			TimestampFormat: time.DateTime,
		}
	default:
		formatter = &logrus.TextFormatter{
			TimestampFormat: time.DateTime,
			FullTimestamp:   true,
		}
	}

	inner.SetFormatter(formatter)
}
```

Comment (hanshal101) [reply backend/impls/logruslog/init.go:61]: Ohk I'll make the necessary changes

**Review by hanshal101 — COMMENTED** [review]
Hello @ferminhg I have made the necessary changes. Please provide if any-other changes need to be made. Hope, this solves the previous problem.

**Review by klesh — APPROVED** [review]

**Comment by klesh** [comment]
Thanks for your contribution

**Comment by JoshuaPoddoku** [comment]
Hi @hanshal101,

Thank you for your contribution! I'm Joshua Poddoku, a committer at Apache DevLake. We'd love for you to join our Slack community or DM me incase you've joined already: https://join.slack.com/t/devlake-io/shared_invite/zt-18uayb6ut-cHOjiYcBwERQ8VVPZ9cQQw. Could you please share your full name there? We want to recognize your efforts with a certificate of contribution.

Best,
Joshua Poddoku

**Pull request merged**